### DISCUSSION TITLE
Inclusion of 'state' times in the report CSV export

### DISCUSSION THREAD

**Discussion by andrewpatto** [CSCfi/rems#3142] [Ideas]
Hi, we are looking at mechanisms for doing some simple reporting out of REMS (possibly with downstream Power BI / Tableau etc).

One of the metrics of interest is some measure of turn around time - how many applications are in each state and how long have they been stuck there. And of course how quickly applications have actually been approved or rejected.

At the moment the export has the time of submission - which is obviously a useful base - but has no other times of transition to other states.

I realise that the transition model means that applications can move between states multiple times - but I wonder if perhaps an export of the 'time in current state' (?)... or columns like 'lastTimeIntoApproved', 'lastTimeIntoRejected' would be possible?

Interested in your thinking of what might be possible here.

(I realise we can fetch this information from the APIs - but we are looking at a process by which an admin user could just easily do an export and then move the CSV directly into a reporting tool - or Excel!)

**Comment by Macroz** [comment]
Hi,

We realize that the reporting functions are very underdeveloped at the moment, as they have not seen much use so far.

As you say using the API or even directly the DB (more unstable as we do not currently guarantee backwards compatibility) are options in many cases, and any kind of report is going to have an API as well. But we do have the specific reporter role meant for users who can access "standard reports" easily from the UI as well. It's just that we have so far not have requests for specific "standard reports".

The processing times have been of interest to our other users too, so that's why we support showing where an application is "going too old" in the Actions table. It would make sense to report these.

The difficulty in the CSV output is to map the possibly complex state transition, and other properties of an application into a static number of columns. The 'lastTimeIntoApproved' or so would work for the times. 

I would be very interested in having an actual use case for the report, then implementing that case. So if you can formulate in a bit more detail what you would like, I can have a go of implementing just that for a start.

**Comment by andrewpatto** [comment]
Hi, just letting you know - I am progressing this at our end! I want to make sure we get some good use cases so we don't waste your time - so am needing to speak to stakeholders here.
